delivery_purchase/models/delivery_carrier.py

```diff
@@ -123,7 +123,12 @@ def purchase_fixed_send_shipping(self, pickings):
     def purchase_base_on_rule_send_shipping(self, pickings):
         res = []
         for p in pickings:
-            carrier = self._match_address(p.partner_id)
+            partner = (
+                p.purchase_id.dest_address_id
+                if "is_dropship" in p._fields and p.is_dropship
+                else p.partner_id
+            )
```

Comment:
No need to check for other fields:

```suggestion
            partner = p.purchase_id.dest_address_id or p.partner_id
```